index.js

```diff
@@ -184,6 +205,69 @@ function addAttribute(props, prop, value, ctx) {
   }
 }
 
+function vueMergeAttributes(props, info, value) {
+  var groupProp = vueGroupAttribute(info, value)
+  var propName = groupProp.name
+  var propValue = groupProp.value
+
+  props[propName] =
+    typeof propValue === 'object'
+      ? Object.assign({}, props[propName] || {}, propValue)
+      : Array.isArray(propValue)
+        ? [].concat(props[propName] || [], propValue)
+        : propValue
+}
+
+function vueGroupAttribute(info, value) {
+  var name = info.property
+  var props = {}
+
+  if (vueUtils.isTopLevel(name)) {
+    props.name = name
+    props.value = value
+  } else {
+    // Nested modules
+    var nestMatch = name.match(vueUtils.nestableRE)
+    if (nestMatch) {
+      var prefix = nestMatch[1]
+      var suffix = name.replace(vueUtils.nestableRE, function(_, $1, $2) {
+        return $2 === '-' ? '' : $2.toLowerCase()
+      })
+
+      var subProp = {}
+      subProp[suffix] = value
+      props.name = prefix
+      props.value = subProp
+    } else if (vueUtils.dirRE.test(info.attribute)) {
+      // Custom directive
+      props.name = 'directives'
+      props.value = [
+        {
+          name: info.attribute.replace(vueUtils.dirRE, ''),
+          value: value
+        }
+      ]
+    } else {
+      // Rest are nested under attrs
+      // guard xlink attributes
+      var attribute = vueUtils.xlinkRE.test(info.attribute)
```

Comment:
the [`info` object](https://github.com/wooorm/property-information) already has information on the space a property is in, `space: 'xlink'` in this case!